src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,782 @@
+
+	public static readonly DependencyProperty VerticalMaxScrollProperty =
+	DependencyProperty.Register(nameof(VerticalMaxScroll), typeof(double), typeof(ZoomContentControl), new PropertyMetadata(0.0d));
+
+	public static readonly DependencyProperty VerticalMinScrollProperty =
+	DependencyProperty.Register(nameof(VerticalMinScroll), typeof(double), typeof(ZoomContentControl), new PropertyMetadata(0.0d));
+
+	public static readonly DependencyProperty HorizontalMaxScrollProperty =
+	DependencyProperty.Register(nameof(HorizontalMaxScroll), typeof(double), typeof(ZoomContentControl), new PropertyMetadata(0.0d));
+
+	public static readonly DependencyProperty HorizontalMinScrollProperty =
+	DependencyProperty.Register(nameof(HorizontalMinScroll), typeof(double), typeof(ZoomContentControl), new PropertyMetadata(0.0d));
+
+	public static readonly DependencyProperty HorizontalScrollValueProperty =
+	DependencyProperty.Register(nameof(HorizontalScrollValue), typeof(double), typeof(ZoomContentControl), new PropertyMetadata(0.0d));
+
+	public static readonly DependencyProperty VerticalScrollValueProperty =
+	DependencyProperty.Register(nameof(VerticalScrollValue), typeof(double), typeof(ZoomContentControl), new PropertyMetadata(0.0d));
+
+	public static readonly DependencyProperty ViewPortWidthProperty =
+	DependencyProperty.Register(nameof(ViewPortWidth), typeof(double), typeof(ZoomContentControl), new PropertyMetadata(0.0d));
+
+	public static readonly DependencyProperty ViewPortHeightProperty =
+	DependencyProperty.Register(nameof(ViewPortHeight), typeof(double), typeof(ZoomContentControl), new PropertyMetadata(0.0d));
+
+	//Not a real margin, but a spacing to avoid overlapped by ShellControlPanel
+	public Thickness AdditionalMargin
+	{
+		get => (Thickness)GetValue(AdditionalMarginProperty);
+		set => SetValue(AdditionalMarginProperty, value);
+	}
+
+	public bool AutoZoomToCanvasOnSizeChanged
+	{
+		get => (bool)GetValue(AutoZoomToCanvasOnSizeChangedProperty);
+		set => SetValue(AutoZoomToCanvasOnSizeChangedProperty, value);
+	}
+
+	public bool IsActive
+	{
+		get => (bool)GetValue(IsActiveProperty);
+		set => SetValue(IsActiveProperty, value);
+	}
+
+	public bool IsZoomAllowed
+	{
+		get => (bool)GetValue(IsZoomAllowedProperty);
+		set => SetValue(IsZoomAllowedProperty, value);
+	}
+
+	public bool IsHorizontalScrollBarVisible
+	{
+		get => (bool)GetValue(IsHorizontalScrollBarVisibleProperty);
+		set => SetValue(IsHorizontalScrollBarVisibleProperty, value);
+	}
+
+	public bool IsVerticalScrollBarVisible
+	{
+		get => (bool)GetValue(IsVerticalScrollBarVisibleProperty);
+		set => SetValue(IsVerticalScrollBarVisibleProperty, value);
+	}
+
+	public BoundsVisibilityData ContentBoundsVisibility
+	{
+		get => (BoundsVisibilityData)GetValue(ContentBoundsVisibilityDataProperty);
+		set => SetValue(ContentBoundsVisibilityDataProperty, value);
+	}
```

Comment:
consider blocking off properties that shouldnt be touched by user:
```suggestion
	public BoundsVisibilityData ContentBoundsVisibility
	{
		get => (BoundsVisibilityData)GetValue(ContentBoundsVisibilityDataProperty);
		private set => SetValue(ContentBoundsVisibilityDataProperty, value);
	}
```